src/main/java/com/google/devtools/build/lib/remote/RemoteSpawnRunner.java

```diff
@@ -639,9 +674,13 @@ private SpawnResult handleError(
       status = Status.EXECUTION_FAILED_CATASTROPHICALLY;
       detailedCode = FailureDetails.Spawn.Code.EXECUTION_FAILED;
       catastrophe = true;
-    } else if (remoteCacheFailed) {
+    } else if (BulkTransferException.allCausedByCacheNotFoundException(exception)) {
+      // At this point, cache evictions that affect uploaded inputs have already been handled.
```

Comment:
@coeuvre This is a change in behavior, but I think it's for the better as it avoids retries that are very unlikely to succeed.